app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from typing import Any
+
+
+def cache(func: Callable) -> Callable:
+    cached_results = {}
+
+    def wrapper(*args) -> Any:
+        key = args
+        if key in cached_results:
+            print("Getting from cache")
+            return cached_results[key]
+        else:
+            result = func(*args)
+            cached_results[key] = result
+            print("Calculating new result")
+            return result
```

Comment:
You can use a single return statement out of the if/else statement instead of two inside it.

Reply:
Fixed. Thanks it really looks better